src/shared/awsagent/roles.go

```diff
@@ -46,8 +49,85 @@ func (a *Agent) GetOtterizeRole(ctx context.Context, namespaceName, accountName
 	return true, role.Role, nil
 }
 
+func (a *Agent) GetOtterizeProfile(ctx context.Context, namespaceName, serviceAccountName string) (found bool, profile *rolesanywhereTypes.ProfileDetail, err error) {
+	a.profileCacheOnce.Do(func() {
```

Comment:
If you can do a little refactor here to make it easier on the eye it will be great